File: android/src/main/java/expo/modules/nativeemojispopup/ExpoNativeEmojisPopupModule.kt
package expo.modules.nativeemojispopup

import expo.modules.kotlin.functions.Coroutine
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoNativeEmojisPopupModule : Module() {
  override fun definition() =
    ModuleDefinition {
      Name("ExpoNativeEmojisPopup")

      AsyncFunction("show") Coroutine { params: Map<String, Any?> ->
        val showParams = ReactionPopupShowParams.fromMap(params)
        ReactionPopupPresenter.show(appContext.currentActivity, showParams)
      }

      AsyncFunction("dismiss") Coroutine { ->
        ReactionPopupPresenter.dismiss()
      }
    }
}

File: ios/ExpoNativeEmojisPopupModule.swift
import ExpoModulesCore

public class ExpoNativeEmojisPopupModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoNativeEmojisPopup")

    AsyncFunction("show") { (params: [String: Any]) async throws -> [String: Any] in
      let showParams = try ReactionPopupShowParams(dictionary: params)
      return try await ReactionPopupOverlayPresenter.shared.show(params: showParams)
    }

    AsyncFunction("dismiss") { () async -> Void in
      await ReactionPopupOverlayPresenter.shared.dismiss()
    }
  }
}
